pkg/internal/apis/config/types.go

```diff
@@ -167,6 +167,8 @@ const (
 	IPTablesMode ProxyMode = "iptables"
 	// IPVSMode sets ProxyMode to iptables
 	IPVSMode ProxyMode = "ipvs"
+	// NoneMode disables kube-proxy
+	NoneMode ProxyMode = "none"
```

Comment:
what is the alternative value of this field?
if `""` implies kube-proxy mode this should be reflected in a comment above.

for the API change, as is, this could be useful if one day kind decides to implement it's own proxy ALA Cilium.
adding another ProxyMode.

but if there are no plans for that and if the demand to skip kube-proxy is not high, the better alternative would have been to configure this via kubeadm's v1beta3 ClusterConfiguration.

of course, a couple of problems with that would be:
- kubeadm v1beta3 is not ready yet, but this exact skip phases feature is tracked [here](https://github.com/kubernetes/kubeadm/issues/2261)
- it would only work with kubeadm / k8s latest

a more flexible feature in kind would be to allow passing extra args to the `kubeadm init/join` commands the same way kubeadm allows passing such to the kubelet / control-plane. as per how cobra, go's CLI works, subsequent flags with the same key would override prior ones.

Reply:
> what is the alternative value of this field?
> if "" implies kube-proxy mode this should be reflected in a comment above.

"" implies iptables mode.

> but if there are no plans for that and if the demand to skip kube-proxy is not high, the better alternative would have been to configure this via kubeadm's v1beta3 ClusterConfiguration.

Once it's ready, we can change the way we disable kube-proxy.

> but if there are no plans for that and if the demand to skip kube-proxy is not high, the better alternative would have been to configure this via kubeadm's v1beta3 ClusterConfiguration.

If we need to pass more flags to kubeadm, then it makes sense. At the moment, it's only one flag.

Reply:
for the API of this field I think it should be
- "iptables" : default value
- "ipvs"
- "none"

if there are other proxies they should provide their installation, same we are doing with the cni

> a more flexible feature in kind would be to allow passing extra args to the kubeadm init/join commands the same way kubeadm allows passing such to the kubelet / control-plane. as per how cobra, go's CLI works, subsequent flags with the same key would override prior ones.

_I think we already discussed this in another issue_ :thinking:

Reply:
flags are indeed the riskier option.